lib/web_ui/lib/src/ui/text.dart

```diff
@@ -1242,7 +1242,15 @@ abstract class Paragraph {
   /// on both sides. In such cases, this method will return [offset, offset+1].
   /// Word boundaries are defined more precisely in Unicode Standard Annex #29
   /// http://www.unicode.org/reports/tr29/#Word_Boundaries
-  List<int> getWordBoundary(dynamic position);
+  TextRange getWordBoundary(TextPosition  position);
+
+  /// Returns the [TextRange] of the line at the given offset.
+  ///
+  /// The newline (if any) is returned as part of the range.
+  ///
+  /// This can potentially be expensive, since it needs to compute the line
```

Comment:
The line metrics are computed as part of layout, so as long as it is used after layout, this should not be particularly expensive. We currently don't do any additional computation beyond that of layout to calculate the LineMetrics.

What should be mentioned though is that this is invalid before layout.

Reply:
Actually, this warning can remain, as it depends on the implementation. Web or SkShaper may have more expensive implementations. It just so happens LibTxt computes this in layout.